Cancer panels

evasterviga opened the pull request:
Priority: High

### Test run
(do it in a screen)

`cd insilico_wgs/validate_wrapper/`

**Sima:** `./validate_wrapper.sh <path-to-your-insilico_wgs-repo>/insilico_panels/KG/brostcancer_lynchs_syndrom_PPAP.v1.0/brostcancer_lynchs_syndrom_PPAP.v1.0.bed KG 1`

**Gwenna:** `./validate_wrapper.sh <path-to-your-insilico_wgs-repo>/insilico_panels/KG/brostcancer_ovarialcancer.v1.0/brostcancer_ovarialcancer.v1.0.bed KG 1
`

### workflow of validate_wrappper.sh: 
(_output dir:_ `/medstore/Development/WGS_validation/in_silico_panels/klinisk_genetik/validate/panels/<panel_name>`)
- run samtools mpileup on every validation sample `--> mpileup/`
- annotate output from samtools mpileup and save separate files for every validation sample `--> mpileup/annotated`
- make some general stats for the panel coverage `--> general_stats`
- convert `mpileup/annotated`-results to lists of regions under 10x or 20x coverage instead of single positions, for every validation sample `--> anybelow/10x/` or `--> anybelow/20x/` **This is where I made changes: ec98f89**
- summarize the low coverage regions into single files `--> anybelow/` **This is where I made changes: 03eccfd**

### Effects of current changes
The changes from ec98f89 makes it possible to save the very last interval discovered from each validation file too. 

If you compare one current output 
`/medstore/Development/WGS_validation/in_silico_panels/klinisk_genetik/validate/panels/kolorektalcancer_NoIntrons.v1.0.bed/anybelow/10x/10x_WGS3_190329_AHJ2JTDMXX_DEDUP_annotated.tsv`
to one prevous:
`/medstore/Development/WGS_validation/in_silico_panels/klinisk_genetik/validate/panels/kolorektalcancer_NoIntrons_tmp_can_be_removed_by_dec2022.v1.0.bed/anybelow/10x/10x_WGS3_190329_AHJ2JTDMXX_DEDUP_annotated.tsv`
You will see one example of a row that was missing before:
`10,89731174,89731183,PTEN,None,None`

The previously mentioned row wasn't included in
`kolorektalcancer_NoIntrons.v1.0.bed/anybelow/kolorektalcancer_NoIntrons.v1.0.bed_10x_sorted.csv`
either, which is what the 03eccfd commit changed.

evasterviga committed:
added gene lists and made bed files

evasterviga committed:
changed for loop to include last anybelow interval from list

evasterviga committed:
changed loop to include last anybelow interval in summary list

Gwennid commented:
Nice explanation, and great that you caught the error! I have a hard time following the different steps of the for loop in `anybelow_sample.py`, but the before/after results you show are convincing. I have a couple of questions/comments (bearing in mind that the description of this PR will help in the future to understand what is going on with the script in general):

- In the description, for the "make some general stats for the panel coverage", it should point at `general_stats`
- Three samples are included in `mpileup` but not in `anybelow` (WGS12, WGS7 and Coriell). Do you know why?
- (Detail) In the fourth point of the description, you could mention the `anybelow/20x` too
- How does the summarizing of regions below 10x/20x happen? Is it regions found in all samples? Or concatenating regions that are found in any sample?

Then I suppose one question is whether we should rerun the coverage analysis for all panels and what we do if we find a difference...

evasterviga commented:
> Nice explanation, and great that you caught the error! I have a hard time following the different steps of the for loop in `anybelow_sample.py`, but the before/after results you show are convincing. I have a couple of questions/comments (bearing in mind that the description of this PR will help in the future to understand what is going on with the script in general):
> 
> * In the description, for the "make some general stats for the panel coverage", it should point at `general_stats`
> * Three samples are included in `mpileup` but not in `anybelow` (WGS12, WGS7 and Coriell). Do you know why?
> * (Detail) In the fourth point of the description, you could mention the `anybelow/20x` too
> * How does the summarizing of regions below 10x/20x happen? Is it regions found in all samples? Or concatenating regions that are found in any sample?
> 
> Then I suppose one question is whether we should rerun the coverage analysis for all panels and what we do if we find a difference...

Great! I understand that the loop was hard to read. I had to read it many times. Also ,I haven't deleted anything from the original, just added rows to not miss any data. Perhaps I could change some variable names to make it a little nicer.

* Yes, It should point at `general_stats`, thank you, I fixed it.
* The samples that aren't in anybelow probably don't have any coverage below for example 10x.
* Sure! I'll mention the 20x as well.
* The10x or 20x regions are collected from all validation samples regardless of how many times they are found. This region for example (`10,89731174,89731183,PTEN,None,None`) is only found in WGS3.

Yes, perhaps we should rerun all the panels. But we should have a talk with the SGs first.

evasterviga committed:
enable rerun without producing changed output - use input and output from separate folders

evasterviga committed:
added condition if currentregion length > 0

sima-r committed:
added convert to float
